# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: netwerk/base/nsURLHelper.cpp
# Commit: c2ca2f596c09
# Full Hash: c2ca2f596c0922e7fd34494dc6ed8dd68d46b122
# Author: abhishekmadan30 <148475204+abhishekmadan30@users.noreply.github.com>
# Date: 2024-03-26 04:11:24
# Regressor Bug: 1877419
# File Overlap Count: 1
# Description:
#   Bug 1877419 - Part 1: Added nested stripping of urls to copy without site tracking. r=pbz,necko-reviewers
#   
#   Depends on D204914
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D202943
# ==============================================================================

diff -r 45931e6db21e -r c2ca2f596c09 netwerk/base/nsURLHelper.cpp
--- a/netwerk/base/nsURLHelper.cpp	Mon Mar 25 18:30:38 2024 +0000
+++ b/netwerk/base/nsURLHelper.cpp	Mon Mar 25 18:30:38 2024 +0000
@@ -1302,9 +1302,7 @@
   });
 }
 
-namespace {
-
-void SerializeString(const nsCString& aInput, nsAString& aValue) {
+void URLParams::SerializeString(const nsCString& aInput, nsAString& aValue) {
   const unsigned char* p = (const unsigned char*)aInput.get();
   const unsigned char* end = p + aInput.Length();
 
@@ -1325,8 +1323,6 @@
   }
 }
 
-}  // namespace
-
 void URLParams::Serialize(nsAString& aValue, bool aEncode) const {
   aValue.Truncate();
   bool first = true;